src/main/webapp/app/components/util/drag-and-drop-question-util.service.js

```diff
@@ -0,0 +1,223 @@
+                            return !isSameDropLocationOrDragItem(dropLocation, mapping.dropLocation);
+                        });
+                        availableDragItems = availableDragItems.filter(function (dragItem) {
+                            return !isSameDropLocationOrDragItem(dragItem, mapping.dragItem);
+                        });
+                    }
+                });
+            }
+
+            // solve recursively
+            var solved = solveRec(question.correctMappings, remainingDropLocations, availableDragItems, sampleMappings);
+
+            if (solved) {
+                return sampleMappings;
+            } else {
+                return [];
+            }
+        }
+
+        /**
+         * Try to solve a drag and drop question recursively
+         *
+         * @param correctMappings {Array} the correct mappings defined by the creator of the question
+         * @param remainingDropLocations {Array} the drop locations that still need to be mapped (recursion stops if this is empty)
+         * @param availableDragItems {Array} the unused drag items that can still be used to map to drop locations (recursion stops if this is empty)
+         * @param sampleMappings {Array} the mappings so far
+         * @return {boolean} true, if the question was solved (solution is saved in sampleMappings), otherwise false
+         */
+        function solveRec(correctMappings, remainingDropLocations, availableDragItems, sampleMappings) {
+            if (remainingDropLocations.length === 0) {
+                return true;
+            }
+
+            var dropLocation = remainingDropLocations[0];
+            return availableDragItems.some(function (dragItem, index) {
+                var correctMapping = getMapping(correctMappings, dragItem, dropLocation);
+                if (correctMapping) {
+                    sampleMappings.push(correctMapping); // add new mapping
+                    remainingDropLocations.splice(0, 1); // remove first dropLocation
+                    availableDragItems.splice(index, 1); // remove the used dragItem
+                    var solved = solveRec(correctMappings, remainingDropLocations, availableDragItems, sampleMappings);
+                    remainingDropLocations.splice(0, 0, dropLocation); // re-insert first dropLocation
+                    availableDragItems.splice(index, 0, dragItem); // re-insert the used dragItem
+                    if (!solved) {
+                        sampleMappings.pop(); // remove new mapping (only if solution was not found)
+                    }
+                    return solved;
+                } else {
+                    return false;
+                }
+            });
+        }
+
+        /**
+         * Check if when there are multiple options for a drop location, you can actually place each of them
+         * and still end up with a 100% correct solution
+         *
```

Comment:
Please add more documentation about this method. As far as I can see, it validates something, so it might make sense to call it "validateAllItemOptionsPossible"